### ISSUE TITLE
package.json does not export ./package.json

### ISSUE THREAD

**Issue by nicksellen** [vuejs/vuex#2160]
### Version
4.0.2

### Reproduction link
[github.com](https://github.com/nicksellen/vuex-package-json)

### Steps to reproduce
Steps are included in the README for the repo, here is a minimal version. I have node version v17.9.0 so it might be dependent on some recent changes there...

```
git clone https://github.com/nicksellen/vuex-package-json.git
cd vuex-package-json
yarn
npx quasar info
```

### What is expected?
It should show that vuex is installed, like this:

```
vuex - 4.0.2 -- state management for Vue.js
```

### What is actually happening?
It shows:

```
vuex - Not installed
```

When I add some debug output, the error underlying is:

```
Package subpath './package.json' is not defined by "exports" in /tmp/foo/quasar-project/node_modules/vuex/package.json
```

---
It's actually fixed in the #main branch of vuex 6 months ago in [this commit](https://github.com/vuejs/vuex/commit/397e9fba45c8b4ec0c4a33d2578e34829bd348d7) , it's just not been released...  manually applying that change to `node_modules/vuex/package.json` resolves the issue locally.

**Comment by mk-pmb** [comment]
I'm affected as well. When will a version with the patch be released?

**Comment by maffmeier** [comment]
I've the same problem.

**Comment by yyx990803** [comment]
fixed by 397e9fb in 4.1
